src/common/tracer.cc

```diff
@@ -42,8 +42,15 @@ jspan Tracer::start_trace(opentelemetry::nostd::string_view trace_name) {
   return noop_tracer->StartSpan(trace_name);
 }
 
+jspan Tracer::start_trace(opentelemetry::nostd::string_view trace_name, bool trace_is_enabled) {
+  if (trace_is_enabled) {
+    return tracer->StartSpan(trace_name);
+  }
+  return noop_tracer->StartSpan(trace_name);
```

Comment:
remind us what the overhead is for noop_tracer?

Reply:
no overhead. just returns a static span